charts/redpanda/templates/rbac.yaml

```diff
@@ -33,9 +33,31 @@ rules:
     - ""
     resources:
       - nodes
+      - secrets
```

Comment:
why secret is added here?

Reply:
In order to get values file we need to get and watch for secrets.

Reply:
I don't understand why Redpanda chart needs to have access to secrets at the Cluster level. 

In the new 2 controllers (nor decommission nor nodeWatcher) https://github.com/redpanda-data/redpanda/pull/12847 I don't see any of them to get nor watch for secrets. 

Maybe I'm missing something with the side car approach. Could you help me understand that?

Reply:
The need does not need to be explicit. If you look closely I use the helm library to get the values file. Helm itself stores stateful information regarding the releases in a secret. This is the chain that explains the need for it.